feature/analytics-Dylan-AddedClickAndToggleEvents

Sparowhawk opened the pull request:
## Description of Change
Added click and toggle events for analytics. Added them to letter list screen,  share app button in settings, buttons that go to edit address(letters and profile), notifications screen toggle and button, sign out button in settings

## Screenshots/Video
N/A

## Testing
Yarn Test

- [x] Tested on iOS <!-- simulator is fine -->
- [x] Tested on Android <!-- simulator is fine -->

## Reviewer Validations
add click and toggle analytic events and apply them to letters list screen when clicking a letter, share app button on settings, buttons that go to edit address(letters and profile), notifications screen toggle and button, sign out button in settings


## PR Checklist
<!-- Engineer: make sure all these items are checked off before requesting a review -->
  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

dumathane reviewed (changes requested):
I'd like to move back to navigateTo here to make our pending code upkeep ticket easier and to drop the navigationRoot stuff if we can?

Sparowhawk commented:
navigateTo will not work
this is every unit test that uses the navigateTo with the () afterwards
TypeError: navigateTo(...) is not a function
it works for navigating
but doesn’t work for the unit tests

Sparowhawk commented on VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/ContactInformationScreen.tsx:
```diff
@@ -123,8 +118,26 @@ const ContactInformationScreen: FC<ContactInformationScreenProps> = ({ navigatio
+      addressType: profileAddressOptions.MAILING_ADDRESS,
+      onPress: () => {
+        logAnalyticsEvent(Events.vama_click(t('contactInformation.mailingAddress'), t('contactInformation.title')))
+        navigationRoot.navigate('EditAddress', {
```
updated

Sparowhawk commented:
> I'd like to move back to navigateTo here to make our pending code upkeep ticket easier and to drop the navigationRoot stuff if we can?

updated and added notes to the upkeep ticket

dumathane commented on VAMobile/src/store/slices/notificationSlice.ts:
```diff
@@ -49,6 +50,7 @@ export const registerDevice =
   (deviceToken?: string, refreshNotificationScreen?: boolean): AppThunk =>
   async (dispatch) => {
     dispatch(dispatchStartRegisterDevice())
+    setAnalyticsUserProperty(UserAnalytics.vama_uses_notifications(deviceToken ? true : false))
```
This isn't the right spot for this, in retrospect I don't think. We'll need to tinker more with where this gets setup. @Sparowhawk hit me tomorrow when you're in about this.

theodur replied:
Should this be called in the [register](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/components/NotificationManager/NotificationManager.tsx#L22) function of the `NotificationManager` instead? `registerDevice` is called in different scenarios, so if we only want to track whether a user uses push notification once during the initial registration, I believe this would need to be moved to the `NotificationManager`

dumathane replied:
Ya, no, Dylan has this correct where it is. I talked it out with him a lot earlier.
